docs/en/query_language/select.md

```diff
@@ -92,23 +92,27 @@ FROM
 ```
 
 
-### FROM Clause
+### FROM Clause {#select-from}
 
 If the FROM clause is omitted, data will be read from the `system.one` table.
-The 'system.one' table contains exactly one row (this table fulfills the same purpose as the DUAL table found in other DBMSs).
+The `system.one` table contains exactly one row (this table fulfills the same purpose as the DUAL table found in other DBMSs).
 
-The FROM clause specifies the table to read data from, or a subquery, or a table function; ARRAY JOIN and the regular JOIN may also be included (see below).
+The `FROM` clause specifies the source to read data from:
 
-Instead of a table, the SELECT subquery may be specified in brackets.
-In this case, the subquery processing pipeline will be built into the processing pipeline of an external query.
-In contrast to standard SQL, a synonym does not need to be specified after a subquery. For compatibility, it is possible to write 'AS name' after a subquery, but the specified name isn't used anywhere.
+- Table
+- Subquery
+- [table function](table_functions/index.md)
+
+`ARRAY JOIN` and the regular `JOIN` may also be included (see below).
 
-A table function may be specified instead of a table. For more information, see the section "Table functions".
+Instead of a table, the `SELECT` subquery may be specified in brackets.
+In this case, the subquery processing pipeline will be built into the processing pipeline of an external query.
```

Comment:
this sentence is especially weird, we should probably rewrite the whole FROM section introduction by explaining possible options and explicitly going through them in order